doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst

```diff
@@ -0,0 +1,300 @@
+
+KMU push operations
+
+Keys with the usage schemes CRACEN_KMU_KEY_USAGE_SCHEME_PROTECTED and CRACEN_KMU_KEY_USAGE_SCHEME_SEED can push data from the slots directly to CRACEN registers that are not accessible by the CPU.
+These usage schemes are supported for pushing symmetric keys used for cipher operations, and for seeds for use by the CRACEN IKG.
+
+Keys with the usage scheme CRACEN_KMU_KEY_USAGE_SCHEME_RAW, such as symmetric keys, must be temporarily pushed to a RAM location, before CRACEN can load them into the asymmetric engine before running operations like sign or verify.
+Key slots that are tagged with CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED also have to be decrypted to a temporary push location in RAM before they are used by CRACEN.
+
+When the application is built with TF-M, this temporary push location is protected inside the secure processing environment to avoid exposing the key material to the non-secure application.
+If TF-M is not used, the keys are pushed to a reserved RAM area at location 0x20000000-0x20000064 (kmu_push_area).
+
+The following special error codes are available:
+
+* PSA_ERROR_INVALID_HANDLE: Attempting an operation is attempted on an empty KMU slot
+* PSA_ERROR_NOT_PERMITTED: Trying an operation on a revoked key
+* PSA_ERROR_HARDWARE_FAILURE: Key slot has invalid data
+* PSA_ERROR_CORRUPTION_DETECTED: Key slot has invalid data
+
+CRACEN IKG
+**********
+
+The Isolated Key Generator (IKG) can derive 3 keys from a 384-bit seed value.
+This seed value can be stored in the KMU, and pushed directly to the IKG without being accessible by the CPU.
+The derived keys are also not accessible by the CPU, but they can be used for cryptographic operations by CRACEN.
+
+These keys are also accessed using normal PSA Crypto APIs, and are referenced by special built-in key IDs.
+
++-----------------+---------------------------------+---------------------------------------------------+
+| Key type        | Key ID                          | Description                                       |
++=================+=================================+===================================================+
+| ECC secp256r1   | CRACEN_BUILTIN_IDENTITY_KEY_ID  | Used for signing/verification.                    |
++-----------------+---------------------------------+---------------------------------------------------+
+| AES 256-bit     | CRACEN_BUILTIN_MKEK_ID          | Used for encryption/decryption or key derivation. |
++-----------------+---------------------------------+---------------------------------------------------+
+| AES 256-bit     | CRACEN_BUILTIN_MEXT_ID          | Used for encryption/decryption or key derivation. |
++-----------------+---------------------------------+---------------------------------------------------+
+
```

Comment:
Maybe also worth mentioning to the reader: 

`CRACEN_BUILTIN_MKEK_ID` and `CRACEN_BUILTIN_MEXT_ID` are master keys and they cannot be used directly. The key derivation algorithm `PSA_ALG_SP800_108_COUNTER_CMAC` should be used to derive keys first. https://arm-software.github.io/psa-api/crypto/1.2/api/ops/kdf.html#c.PSA_ALG_SP800_108_COUNTER_CMAC.

Reply:
Reading this section again, it seems to just be mentioned to the user for informational purposes and end-users aren't actually meant to use these keys in their code. If that is the case, we don't need to add this extra piece of info.

Reply:
We'll leave this section as it is, although with a note that this is not really meant for end-users to be using directly.